### PR TITLE
add(support): toit programming language

### PR DESCRIPTION
**Pull request by snxx-lppxx**

### CODE REVIEW

File: etc/config/toit.defaults.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+compilers=/usr/bin/toit
```
Comment (RubenRBS) [inline etc/config/toit.defaults.properties:1]: Is this meant to be also live in the site? if so, it's missing some things:
- A PR in the sister repo https://github.com/compiler-explorer/infra that installs the compiler to the required places (There are some example PRs you can check out)
- And a `toit.amazon.properties` files to configure the compilers for the live site

Also, you should add the language to `.github/labeler.yml`

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Hi! Thanks for this :) I have some requests and questions, but overall looks great

File: lib/compilers/toit.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
```
[the remark this answers is not in the document]
Comment (RubenRBS) [reply lib/compilers/toit.js:1]: Also, before I forget. We're trying to slowly move towards typescript, so we would appreciate if any new files could be written in TS, which _should_ be easy enough, but do let us know if you have any questions regarding that :)

File: etc/config/toit.amazon.properties [hunk 3]
```diff
@@ -0,0 +1,10 @@
+compilers=&toit
+defaultCompiler=toit16
+
+group.toit.compilers=toit16
+group.toit.isSemVer=true
+group.toit.baseName=Toit
+group.toit.compilerType=toit
+
+compiler.toit161.semver=1.6
+compiler.toit161.exe=/opt/compiler-explorer/toit-1.6/bin/toit
```
Comment (partouf) [inline etc/config/toit.amazon.properties:10]: the executable is installed with the name `toit.run`

**Review by partouf — CHANGES_REQUESTED** [review]

**Comment by partouf** [comment]
Sidenote; you can add your original Megagreater example to the folder `examples/toit` so that it can be loaded in by users

**Comment by partouf** [comment]
Thanks for the addition. I have disabled Toit for the live website as there is no way to show assembly or IR atm, but it's available for users that run a local installation, as long as they select "Execute the code"

**Pull request closed without merging**